daikon-schema/daikon-dataset-schema/src/main/resources/dataset/datasetFieldTypeSchema.json

```
@@ -0,0 +1,63 @@
+{
+  "$schema": "https://json-schema.org/draft/2019-09/schema",
+  "$id": "https://org.talend.daikon/dataset.field.type.schema.json",
+  "title": "Dataset field Type",
+  "description": "A type of field of a dataset on Talend ecosystem",
+  "type": "object",
+  "properties": {
+    "fields": {
+      "description": "children fields",
+      "type": "array",
+      "items": {
+        "$ref": "https://org.talend.daikon/dataset.field.schema.json"
+      }
+    },
+    "type": {
+      "description": "primitive type of the field",
+      "type": "string"
+    },
+    "description": {
+      "description": "Description of the field",
+      "type": "string"
+    },
+    "dqType": {
+      "description": "DQ type",
+      "type": "string"
+    },
+    "dqTypeId": {
+      "description": "DQ type ID",
+      "type": "string"
+    },
+    "dqNativeType": {
+      "description": "DQ native type",
+      "type": "string"
+    },
+    "isForced": {
+      "description": "Is DQ type force on this field",
+      "type": "boolean",
+      "default": false
+    },
+    "isNativeForced": {
+      "description": "Is DQ type force on this field",
+      "type": "boolean",
+      "default": false
+    },
+    "talend.component.semanticType": {
+      "description": "DQ semantic type matching result",
+      "$ref": "https://org.talend.daikon/dq.type.match.schema.json"
+    },
+    "talend.component.dqType": {
+      "description": "DQ native type matching result",
+      "$ref": "https://org.talend.daikon/dq.type.match.schema.json"
+    },
+    "talend.component.label": {
+      "description": "Real label of the field",
```

what is a Real label ?

---

Real label is existing label as it is defined in a CSV/database etc. Unlike the avro `name` field It is not unique and can contains special characters

---

Real label sounds wierd for the description, may be just what you wrote above would be more meaningfull like
```suggestion
      "description": "Actual field or column name as defined in the original file or database",
```

---

You must know that in the Talend Studio there are 3 "label" for a field
name : <columnName modified / unically identifying the field like column0,column1 for not compatible names...) ==> (avro name)
ColumnName: name of the column compatible with Java code for the Code generation
DBColumnName: Original field name (can even be japanes names)

The ColumnName is not really relevant here as we don't care about Java code génération but this just to be aware of it as we will eventually have TC dataset being used in the Studio (Dizy)

---

I'd vote for `OriginalFieldName` similar to [Schema](https://github.com/Talend/component-runtime/blob/master/component-api/src/main/java/org/talend/sdk/component/api/service/schema/Schema.java#L123) in component-runtime.